force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,193 @@
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
+            tasksToInsert.add(tsk);
+        }
+        insert tasksToInsert;
+    }
+
+    /*
+    * setPrimaryContact()
+    * Before Update
+    * When an opportunity is updated set the primary contact on the opportunity to the contact with the title of 'CEO'.
+    */
+    public void setPrimaryContact(){
+        List<Id> accountIds = new List<Id>();
+        
+        for(Opportunity opp : triggerList){
+            accountIds.add(opp.AccountId);
+        }
+
+        List<Contact> ceoList = [SELECT Id, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO'];
+
+        Map<Id, Id> accIdConIdMap = new Map<Id, Id>();
+
+        for(Contact con : ceoList){
+            accIdConIdMap.put(con.AccountId,con.Id);
+        }
+
+        for(Opportunity opp : triggerList){
+            opp.Primary_Contact__c = accIdConIdMap.get(opp.AccountId);
+        }
+    }
+
+    /*
+    * amountValidation()
+    * When an opportunity is updated validate that the amount is greater than 5000.
+    * 
+    */
+    public void amountValidation(){
+        for(Opportunity opp : triggerList){
+            if(opp.Amount <= 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+    /*
+    * updateDescriptionFromStage()
+    * When an opportunity is updated, the stage name is written in the description
+    */
+    public void updateDescriptionFromStage(){
+        for(Opportunity opp : triggerList){
+            opp.Description = 'Stage Change:' + opp.StageName;            
+        }
+    }
+
+    /*
+    * notifyOwnersOpportunityDeleted()
+    * Sends an email notification to the owner of the Opportunity when it gets deleted.
+    * Uses Salesforce's Messaging.SingleEmailMessage to send the email.
+    */
+    public void notifyOwnersOpportunityDeleted() {
+        List<Messaging.SingleEmailMessage> mails = new List<Messaging.SingleEmailMessage>();
+        for (Opportunity opp : triggerList){
+            Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+            String[] toAddresses = new String[] {[SELECT Id, Email FROM User WHERE Id = :opp.OwnerId].Email};
+            mail.setToAddresses(toAddresses);
+            mail.setSubject('Opportunity Deleted : ' + opp.Name);
+            mail.setPlainTextBody('Your Opportunity: ' + opp.Name +' has been deleted.');
+            mails.add(mail);
+        }        
+        
+        try {
+            Messaging.sendEmail(mails);
+        } catch (Exception e){
+            System.debug('Exception: ' + e.getMessage());
+        }
+    }
+
+    public void validateCloseOpportunity(){
+        for(Opportunity opp : triggerList){
+            if(opp.StageName.contains('Closed')){
```

Comment:
There is an IsClosed field on Opportunity that is probably safer to use rather than the string.